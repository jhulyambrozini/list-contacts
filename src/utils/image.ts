const getRandomColor = (): string => {
  const red = Math.floor(Math.random() * 256)
  const green = Math.floor(Math.random() * 256)
  const blue = Math.floor(Math.random() * 256)

  return `rgb(${red},${green},${blue})`
}

export const generateRandomImage = (
  text: string,
  width: number,
  height: number
): string => {
  const canvas = document.createElement('canvas')
  canvas.width = width
  canvas.height = height

  const ctx = canvas.getContext('2d')
  if (!ctx) {
    throw new Error('Canvas 2D context is not supported')
  }

  // Preencher o canvas com uma cor aleatória
  ctx.fillStyle = getRandomColor()
  ctx.fillRect(0, 0, width, height)

  // Configurar o texto no centro
  ctx.fillStyle = 'white'
  ctx.font = '50px Arial'
  ctx.textAlign = 'center'
  ctx.textBaseline = 'middle'

  // Capitalizar a primeira letra da string e colocá-la no centro
  const firstLetter = text.charAt(0).toUpperCase()
  ctx.fillText(firstLetter, width / 2, height / 2)

  // Converter o canvas para um formato de dados URL
  const dataURL = canvas.toDataURL('image/png')
  return dataURL
}

export const validateImageFormat = (file: File): boolean => {
  const allowedFormats = ['image/png', 'image/jpeg']
  return allowedFormats.includes(file.type)
}
