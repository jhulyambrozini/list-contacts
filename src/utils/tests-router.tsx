import { ReactNode } from 'react'
import { BrowserRouter, useNavigate } from 'react-router-dom'
import { render } from '@testing-library/react'

const renderWithRouter = (component: ReactNode) => {
  let navigate: any
  const Wrapper = () => {
    navigate = useNavigate()
    return component
  }

  return {
    ...render(
      <BrowserRouter>
        <Wrapper />
      </BrowserRouter>
    ),
    navigate
  }
}

export default renderWithRouter
