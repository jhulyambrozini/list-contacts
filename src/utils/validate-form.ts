import { FormikProps } from 'formik'
import { ValuesForm } from '../types/Form'

export const getFieldStatus = (
  field: string,
  form: FormikProps<ValuesForm>
) => {
  const isTouched = field in form.touched
  const isInvalid = field in form.errors
  return { isTouched, isInvalid }
}

export const getMessageError = (
  field: string,
  form: FormikProps<ValuesForm>,
  message?: string
) => {
  const { isTouched, isInvalid } = getFieldStatus(field, form)
  return isTouched && isInvalid ? message : false
}

export const getSuccess = (field: string, form: FormikProps<ValuesForm>) => {
  const { isTouched, isInvalid } = getFieldStatus(field, form)
  return isTouched && !isInvalid
}
