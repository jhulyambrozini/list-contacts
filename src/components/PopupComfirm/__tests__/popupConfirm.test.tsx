import '@testing-library/jest-dom'

import { fireEvent, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { vi } from 'vitest'

import PopupConfirm from '..'
import { renderWithProvider } from '../../../utils/tests-redux'

const renderComponent = () => {
  const { store, container } = renderWithProvider(
    <MemoryRouter>
      <PopupConfirm id="1" />
    </MemoryRouter>,
    {
      preloadedState: {
        popUp: {
          isPopupOpen: true
        },
        contact: {
          items: [
            {
              email: 'exemple@gmail.com',
              id: '1',
              image: '/img.png',
              firstName: 'Kamilly',
              lastName: 'Silva',
              tel: '11 99999-9999'
            }
          ]
        }
      }
    }
  )
  return { store, container }
}

const mockUseNavigate = vi.fn()

vi.mock('react-router-dom', async () => {
  const actual = (await vi.importActual('react-router-dom')) as any
  return {
    ...actual,
    useNavigate: () => mockUseNavigate
  }
})

describe('<PopupConfirm />', () => {
  it('should render correctly when isPopupOpen is true', () => {
    const { container } = renderComponent()

    expect(
      screen.getByRole('dialog', { name: /excluir contato/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should close popup when clicking on no', () => {
    const { store } = renderComponent()
    const noButton = screen.getByRole('button', { name: /nÃ£o/i })

    fireEvent.click(noButton)

    const state = store.getState()

    expect(state.popUp.isPopupOpen).toBeFalsy()
  })

  it('should call the removeContact function when clicking on yes', () => {
    const { store } = renderComponent()
    const yesButton = screen.getByRole('button', { name: /sim/i })

    fireEvent.click(yesButton)

    const state = store.getState()

    expect(state.popUp.isPopupOpen).toBeFalsy()
    expect(state.contact.items.length).toEqual(0)
    expect(mockUseNavigate.mock.calls[0][0]).toBe('/')
  })
})
