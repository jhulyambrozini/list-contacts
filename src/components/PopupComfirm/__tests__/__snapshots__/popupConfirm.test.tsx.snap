// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<PopupConfirm /> > deve renderizar corretamente quando ispopupopn for true 1`] = `
<div
  aria-description="Pop up para confirmação da exclusão do contato"
  aria-label="Excluir contato"
  class="sc-eDnVMP mcwsR"
  role="dialog"
>
  <p>
    Deseja excluir?
  </p>
  <div>
    <button
      class="sc-bcPKhP sc-dkjKgF sVHel jtVglq"
      padding="1.2rem 2.4rem"
      title="Não excluir contato"
      type="button"
    >
      <span>
        NÃO
      </span>
    </button>
    <button
      class="sc-bcPKhP sc-dkjKgF sVHel jtVglq"
      padding="1.2rem 2.4rem"
      title="Confirmar excluir contato"
      type="button"
    >
      <span>
        SIM
      </span>
    </button>
  </div>
</div>
`;
